src/fable/Fable.Core/npm/fable-core.ts

```diff
@@ -943,16 +943,22 @@ class FString {
     return FString.padLeft(str, len, ch, true);
   }
 
-  static remove(str: string, args1: number, args2?: number) {
-    var start = 0;
-    var len = 0;
-    if (args2 === undefined) {
-      len = args1;
+  static remove(str: string, startIndex: number, count?: number) {
+    if (count === undefined) {
+      if (startIndex >= str.length) {
+        throw "startIndex must be less than length of string";
+      }
+    } else {
+      if (startIndex >= str.length || count > str.length) {
+        throw "Index and count must refer to a location within the string."
+      }
+    }
+
+    if (count === undefined) {
```

Comment:
I usually use `count == null` because it covers both `null` and `undefined` (note it's using non-strict equality `==`) which are semantically the same for Fable. Here it would be even safer `typeof count === "number"`, but see below.